cl/recap/tasks.py

```diff
@@ -3435,3 +3437,89 @@ def do_recap_document_fetch(epq: EmailProcessingQueue, user: User) -> None:
         process_recap_email.si(epq.pk, user.pk),
         extract_recap_pdf.s(),
     ).apply_async()
+
+
+def _run_availability_check(rd: RECAPDocument) -> bool:
```

Comment:
this function should be implemented as a task. It allows for clear decoupling of the court page validation logic from the main request flow. Additionally, this refactoring creates a natural point for integrating email notification functionality (the prayer_unavailable method)